frontend/tests/controllers/UserRegistrationTest.php

```diff
@@ -116,4 +116,75 @@ public function testUserLoggedViaGoogleAndThenNativeModeWithDifferentUsername()
         }
     }
+
+     /**
+     * user13 logged in and a parental Token is generated
+     *
+     */
+    public function testUser13ToGenerateParentalTokenAtTimeOfRegistration() {
+        // Verify that the token is generated.
+        $under13BirthDateTimestamp = strtotime('-10 years');
+        $randomString = \OmegaUp\Test\Utils::createRandomString();
+        \OmegaUp\Controllers\User::apiCreate(
+            new \OmegaUp\Request([
+                'username' => $randomString,
+                'password' => $randomString,
+                'parent_email' => $randomString . '@' . $randomString . '.com',
+                'birth_date' => $under13BirthDateTimestamp,
+            ]),
+            $this->assertNotNull($under13BirthDateTimestamp)
+        );
+        $response = \OmegaUp\DAO\Users::FindByUsername($randomString);
+
+        $this->assertNotNull($response->parental_verification_token);
+    }
+
+    /**
+     * user logged in and a parental Token is not generated
+     *
+     */
+    public function testUserDoToGenerateParentalTokenAtTimeOfRegistration() {
+         //Verify that the token is not generated.
+         $over13BirthDateTimestamp = strtotime('-15 years');
+         $randomString = \OmegaUp\Test\Utils::createRandomString();
+        \OmegaUp\Controllers\User::apiCreate(
+            new \OmegaUp\Request([
+                 'username' => $randomString,
+                 'password' => $randomString,
+                 'email' => $randomString . '@' . $randomString . '.com',
+                 'birth_date' => $over13BirthDateTimestamp,
+            ]),
+            $this->assertNotNull($over13BirthDateTimestamp)
+        );
+
+         $response = \OmegaUp\DAO\Users::FindByUsername($randomString);
+
+         $this->assertNull($response->parental_verification_token);
+    }
+
+    /**
+     * user logged in and a parantal token fails due to both email address
+     *
+     */
+    public function testUserParentaltokenNotGeneratedDueEmailAlreadyExists() {
+        $over13BirthDateTimestamp = strtotime('-15 years');
+        $randomString = \OmegaUp\Test\Utils::createRandomString();
+        try {
+            $r = new \OmegaUp\Request([
+                    'username' => $randomString,
+                    'password' => $randomString,
+                    'email' => $randomString . '@' . $randomString . '.com',
+                    'parent_email' => $randomString . '@' . $randomString . '.com',
+                    'birth_date' => $over13BirthDateTimestamp,
+            ]);
+            $this->assertNotNull($over13BirthDateTimestamp);
+
+             $response = \OmegaUp\Controllers\User::apiCreate($r);
+            $this->fail(
+                'User should have not been able to be created because the email already exists in the data base'
+            );
+        } catch (\OmegaUp\Exceptions\DuplicatedEntryInDatabaseException $e) {
+            $this->assertSame('mailInUse', $e->getMessage());
```

Comment:
Actually, this is the exception thrown:
```suggestion
        } catch (\OmegaUp\Exceptions\InvalidParameterException $e) {
            $this->assertSame('parameterInvalid', $e->getMessage());
```